newrelic-agent/src/test/java/com/newrelic/agent/errors/ErrorDataImplTest.java

```diff
@@ -0,0 +1,194 @@
+package com.newrelic.agent.errors;
+
+
+import com.newrelic.agent.Transaction;
+import com.newrelic.agent.TransactionData;
+import com.newrelic.agent.attributes.AttributeNames;
+import com.newrelic.agent.transaction.PriorityTransactionName;
+import com.newrelic.agent.transaction.TransactionThrowable;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+public class ErrorDataImplTest {
```

Comment:
It is a good idea to add tests for when we are in a non-web transaction and you call `getRequestUri `, `getHttpStatusCode `, and `getHttpMethod ` where we expect to return empty string. I suspect the current implementation as of this commit risks a `NullPointerException`.